search-filter.lua

```diff
@@ -0,0 +1,108 @@
+-- Required libraries
+local gui = require('gui') -- Importing the 'gui' library
+local widgets = require('gui.widgets') -- Importing the 'widgets' library
+
+-- Define SearchEngine class
+SearchEngine = defclass(SearchEngine, widgets.Window) -- Defining a new class 'SearchEngine' that inherits from 'widgets.Window'
+SearchEngine.ATTRS = {
+    frame_title='Search Engine for Fort-Controlled Units', -- Title of the frame
+    frame={w=50, h=45}, -- Dimensions of the frame
+    resizable=true, -- Frame can be resized
+    resize_min={w=43, h=20}, -- Minimum dimensions when resizing
+}
+
+function SearchEngine:init() -- Initialization function for the SearchEngine class
+    self:addviews{
+        widgets.EditField{
+            view_id='edit',
+            frame={t=1, l=1}, -- Position of the EditField view
+            text='', -- Initial text in the EditField view
+            on_change=self:callback('updateList'), -- Callback function when text in EditField changes
+        },
+        widgets.List{
+            view_id='list',
+            frame={t=3, b=0},
+            choices=self:getFortControlledUnits(), -- Choices in the List view are obtained from getFortControlledUnits function
+            on_select=self:callback('onSelect'), -- Callback function when a unit is selected from the list
+        }
```

Comment:
This should be a `widgets.FilteredList`

Reply:
Neat. Idk how to properly change it on the git hub or the code. I'm ok with the code being taken over by someone else.